runtime/codert_vm/cnathelp.cpp

```diff
@@ -748,7 +748,90 @@ old_slow_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 void* J9FASTCALL
 old_fast_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	OLD_JIT_HELPER_PROLOGUE(1);
+	DECLARE_JIT_PARM(j9_object_t *, arrayObject, 1);
+	DECLARE_JIT_INT_PARM(index, 1);
+	void *slowPath = NULL;
+	if (fast_jitLoadFlattenableArrayElementImpl(currentThread, arrayObject, index)) {
+		slowPath = (void*) old_slow_jitLoadFlattenableArrayElement;
+	}
+	return slowPath;
+}
+
+void* J9FASTCALL
+old_slow_jitLoadFlattenableArrayElementImpl(J9VMThread *currentThread)
+{
+	SLOW_JIT_HELPER_PROLOGUE();
+	J9Class *arrayref = (J9Class*)currentThread->floatTemp1;
+	if (NULL == arrayref) {
+		rc = THROW_NPE;
```

Comment:
when throwing exceptions you need to set it to the correct exception routine. Examples are:

`addr = setHeapOutOfMemoryErrorFromJIT(currentThread);`
`addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGINSTANTIATIONERROR | J9_EX_CTOR_CLASS, J9VM_J9CLASS_TO_HEAPCLASS(objectClass));`